webpack.config.js

```diff
@@ -29,15 +34,83 @@ module.exports = {
             enforce: "pre",
             test: /\.js$/,
             loader: "source-map-loader"
+        },
+        {
+            test: /\.css$/,
+            use: [
+              MiniCssExtractPlugin.loader,
+              'css-loader'
+            ]
         }
     ]
   },
   resolve: {
     extensions: ['*', '.js', '.jsx', '*.ts', '*.tsx']
   },
   output: {
-    path: __dirname + '/backwardcompatibilityml/widget/resources',
+    path: path.resolve(__dirname, 'backwardcompatibilityml/widget/resources'),
     publicPath: '/',
     filename: 'widget-build.js'
   }
 };
+
+const devConfig = {
```

Comment:
Do we have a different build target here because we want to inject hot reload?

Reply:
If that is the case, would it be possible to keep the same build target, but add the hot reload if we build in development mode?

Reply:
Yes, the general intent is to allow the hot loader to work. The problem I encountered was that the dev mode widget needs some extra stuff compiled into it. What we end up getting are 2 different build artifacts, much like how debug symbols are included in the build with other languages.

`widget-build-dev.js` has the following differences

- React hot loader compiled in
- CSS styles compiled into the JS (which enables hot loading of CSS)

It might be possible to parse command line arguments here and conditionally change up the build depending on that, but that seemed trickier than just having 2 different targets. The current setup will also allow you to test your changes in parallel with Jupyter and the React environment without any need to change up your workflow - just run `npx webpack` to build your latest changes into `widget-build.js`. There won't be any risk of `widget-build.js` and `widget-build-dev.js` conflicting with each other.

Let me know what you think - I'm open to trying to consolidate the build targets if you think it's worthwhile long term.

Reply:
Ok this makes a lot of sense now.

Could we separate out the artifacts into different folders?

So have a top level folder called "development-artifacts" that collect all the build artifacts related to the development server. And drop the `widget-build.js` in there.

Reply:
That's a good idea. Initially I wasn't sure how to separate everything, but that should be doable now.